scanpy/preprocessing/_simple.py

```diff
@@ -859,14 +869,17 @@ def scale_array(
     mean, var = _get_mean_var(X)
     std = np.sqrt(var)
     std[std == 0] = 1
-    if issparse(X):
-        if zero_center:
+    if zero_center:
+        if issparse(X):
             raise ValueError("Cannot zero-center sparse matrix.")
-        sparsefuncs.inplace_column_scale(X, 1 / std)
-    else:
-        if zero_center:
-            X -= mean
-        X /= std
+        if isinstance(X, DaskArray) and issparse(X._meta):
+            warnings.warn(
+                "zero-center being used with `DaskArray` sparse chunks.  This can be bad if you have large chunks or intend to eventually read the whole data into memory."
```

Comment:
The reason I did not do this is because the matrix could be further reduced downstream.  With out-of-core processing, couldn't it be desirable to let this operation go through, especially if they end up using `X` in a way that reduces its dimensions considerably?

Reply:
I think it's okay to request a `map_blocks` to dense before that. I think it's bad to let people create a completley full sparse array. E.g. the error message could say:

"To enable zero centering, first densify the array like: `adata.X = adata.X.map_blocks(lambda x: x.toarray(), dtype=adata.X.dtype)`

Reply:
So this should error then?

Reply:
I would go for erroring, since that's what we do for the in-memory case.

Reply:
Actually, this is a bit of misdirection - we actually densify! https://github.com/scverse/scanpy/blob/14555ba48537995acaa381b8b6ad5fc41e612510/scanpy/preprocessing/_simple.py#L892-L898

Reply:
I am actually not sure why that `issparse` check is here even - this is in the dispatch only for `np.ndarray`!